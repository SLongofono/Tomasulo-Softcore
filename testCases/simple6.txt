  # of rs Cycles in EX  Cycles in Mem # of FUs
Integer adder 1 1   1
FP adder  1 4   1
FP multiplier 1 4    1
Load/store unit 1 1 4 1
ROB entries = 128
CDB buffer entries = 2
R1=1,R2=2,R3=3,F1=1.1,F2=2.1
Mem[0]=0.0,Mem[4]=1.1,Mem[8]=2.2,Mem[12]=3.3

LD F3, 0(R0)
LD F4, 4(R0)
LD F5, 8(R0)
LD F6, 12(R0)
SD F1, 0(R0)
SD F2, 4(R0)
ADDD.D F7,F3,F4
ADDD.D F8,F3,F4
MULT.D F9,F3,F4
MULT.D F10,F3,F4
SUB R9,R3,R4
SUB R10,R3,R4
