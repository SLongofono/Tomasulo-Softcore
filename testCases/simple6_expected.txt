================================Instruction Completion Table====================
ID	| IS		 EX		 MEM		 WB		 COM
0	| 0		 1		 2		 6		 7
1	| 7		 8		 9		 13		 14
2	| 14		 15		 16		 20		 21
3	| 21		 22		 23		 27		 28
4	| 28		 29		 30		 34		 35
5	| 35		 35		 37		 41		 42
6	| 36		 37		 None		 42		 43
7	| 43		 43		 None		 48		 49
8	| 44		 45		 None		 49		 50
9	| 50		 51		 None		 55		 56
10	| 51		 52		 None		 53		 57
11	| 54		 55		 None		 56		 58

================================Integer ARF=====================================
R0 : 0              R1 : 1              R2 : 2              R3 : 3             
R4 : 0              R5 : 0              R6 : 0              R7 : 0              
R8 : 0              R9 : 3              R10: 3              R11: 0              
R12: 0              R13: 0              R14: 0              R15: 0              
R16: 0              R17: 0              R18: 0              R19: 0              
R20: 0              R21: 0              R22: 0              R23: 0              
R24: 0              R25: 0              R26: 0              R27: 0              
R28: 0              R29: 0              R30: 0              R31: 0              

================================Floating Point ARF==============================
F0 : 0.000000                           F1 : 1.100000                           
F2 : 2.100000                           F3 : 0.000000                           
F4 : 1.100000                           F5 : 2.200000                           
F6 : 3.300000                           F7 : 1.100000                           
F8 : 1.100000                           F9 : 0.000000                           
F10: 0.000000                           F11: 0.000000                           
F12: 0.000000                           F13: 0.000000                           
F14: 0.000000                           F15: 0.000000                           
F16: 0.000000                           F17: 0.000000                           
F18: 0.000000                           F19: 0.000000                           
F20: 0.000000                           F21: 0.000000                           
F22: 0.000000                           F23: 0.000000                           
F24: 0.000000                           F25: 0.000000                           
F26: 0.000000                           F27: 0.000000                           
F28: 0.000000                           F29: 0.000000                           
F30: 0.000000                           F31: 0.000000                           


================================Memory Unit=====================================
Word 00: 1.100000                       Word 01: 2.200000                       
Word 02: 2.200000                       Word 03: 3.300000                       
